R/splitByRanks.R

```diff
@@ -121,28 +115,48 @@ setGeneric("splitByRanks",
     SimpleList(ans)
 }
 
-#' @rdname splitByRanks
+#' @rdname agglomerate-methods
 #' @export
-setMethod("splitByRanks", signature = c(x = "SummarizedExperiment"),
-    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, ...){
+setMethod("agglomerateByRanks", signature = c(x = "SummarizedExperiment"),
+    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, as.list = TRUE, ...){
+        browser()
         args <- .norm_args_for_split_by_ranks(na.rm = na.rm, ...)
-        .split_by_ranks(x, ranks, args)
+        if( !.is_a_bool(as.list) ){
+            stop("'as.list' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if( as.list ){
+            .split_by_ranks(x, ranks, args)
+        }
+        else{
+            altExps(x) <- .split_by_ranks(x, ranks, args)
+            x
+        }
```

Comment:
SummarizedExpriment does not have altExps slot. You could convert it to TreeSE and give warning.

Also: you are replacing existing altExps with new list (old alternative experiments are lost). Just add these new ones to the list